main/Main.hs

```diff
@@ -125,10 +125,11 @@ main = do
     Catch connWorker
     ) Nothing
 
-  -- Re-read the config on SIGUSR2
-  void $ installHandler sigUSR2 (
-    Catch $ reReadConfig (cliPath opts) refConf
-    ) Nothing
+  -- Re-read the config on SIGUSR2, but only if we have a config file
```

Comment:
So by using env vars we lose the ability to reload configs without downtime right?

According to https://unix.stackexchange.com/a/314913/278242:

> The environment is copied into the process' address space when the process is created. Changes in the environment only affect newly started processes, you can't change the environment of a running program from the outside.

So we should discourage using them for systemd services. Or should they only be used for docker?

Reply:
I'm not sure whether we should discourage using them, but yes I think the combinations of systemd / config file and docker / environment variables make a lot of sense.

In a docker based setup I would want to use in-database-configuration for those parts that should be reloadable.

Reply:
Right. Discouraging is not needed if we can be clear about reloading on the docs. So no problem here.